Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -283,7 +283,7 @@ open class RouteController: NSObject {
         if (newLegIndex != progress.legIndex) {
             progress.legIndex = newLegIndex
         }
-        if (newStepIndex != progress.currentLegProgress.stepIndex) {
+        if (newStepIndex != progress.currentLegProgress.stepIndex) && (newStepIndex < progress.currentLeg.steps.endIndex - 1) {
```

Comment:
Since `RouteLegProgress.currentStep` is a computed property, which accesses array by index, which might be out of bounds would it work if we check if index is valid right before that? It'd be also beneficial to create a unit test for such use case.